Tests/Realm.Tests/Database/AsymmetricObjectTests.cs

```diff
@@ -0,0 +1,382 @@
+                {
+                    Assert.That(TestHelpers.GetPropertyValue(doc, propertyName), Is.EqualTo(propertyValue));
+                }
+            });
+        }
+
+        [Test]
+        public void MixAddingObjectAsymmetricAndNot()
+        {
+            SyncTestHelpers.RunBaasTestAsync(async () =>
+            {
+                var partitionLike = Guid.NewGuid().ToString();
+                var id = new Random().Next();
+                var flxConfig = await GetFLXIntegrationConfigAsync();
+                flxConfig.Schema = new[] { typeof(BasicAsymmetricObject), typeof(PrimaryKeyInt32Object) };
+
+                flxConfig.PopulateInitialSubscriptions = (realm) =>
+                {
+                    var query = realm.All<PrimaryKeyInt32Object>().Where(n => n.Id == id);
+                    realm.Subscriptions.Add(query);
+                };
+
+                using var realm = await GetFLXIntegrationRealmAsync(flxConfig: flxConfig);
+
+                Assert.DoesNotThrow(() =>
+                {
+                    realm.Write(() =>
+                    {
+                        realm.Add(new BasicAsymmetricObject
+                        {
+                            PartitionLike = partitionLike
+                        });
+
+                        realm.Add(new PrimaryKeyInt32Object
+                        {
+                            Id = id
+                        });
+                    });
+                });
+            });
+        }
+
+        [Test, NUnit.Framework.Explicit("Once Daniel Tabacaru's work is done on \"error actions\" this will be an \"application bug\" action: https://github.com/10gen/baas/blob/9f32d54aa79aff6dfb36a6c07742594e38b07441/realm/sync/protocol/protocol_errors.go#L439")]
+        public void AsymmetricObjectInPbs_Throws()
+        {
+            SyncTestHelpers.RunBaasTestAsync(async () =>
+            {
+                var config = await GetIntegrationConfigAsync();
+                config.Schema = new[] { typeof(BasicAsymmetricObject) };
+                var tcs = new TaskCompletionSource<object>();
+
+                config.OnSessionError = (session, error) =>
+                {
+                    Assert.That(error, Is.InstanceOf<SessionException>());
+
+                    // Assert.That(error.ErrorCode, Is.EqualTo(ErrorCode.ApplicationBug));
+                };
+
+                using var realm = await GetRealmAsync(config);
+
+                await tcs.Task;
+            });
+        }
+
+        [Test]
+        public void AsymmetricObjectInLocalRealm_Throws()
+        {
+            TestHelpers.RunAsyncTest(async () =>
+            {
+                var config = (RealmConfiguration)RealmConfiguration.DefaultConfiguration;
+                config.Schema = new[] { typeof(BasicAsymmetricObject) };
+
+                Exception ex = null;
+                try
+                {
+                    using var realm = await GetRealmAsync(config);
```

Comment:
Why do you need the async version here - wouldn't you get the same error with `GetRealm`?

Reply:
Wouldn't this be true for almost any test? Is there a technical reason you're asking this question on this specific test?
And in general, unless one is testing the async API, what would be the reason to want to specifically use one (the async one) over the other (the sync one)?
(I assume you're asking here because there's no need to use the async method to grab the configuration)